src/analysis.ts

```diff
@@ -273,12 +275,26 @@ export async function analyzeGit(
   const analysisData = await analyzeBundle(baseURL, sessionToken, includeLint, severity, bundleId);
 
   // Create bundle instance to handle extensions
-  return {
-    baseURL,
-    sessionToken,
-    includeLint,
-    severity,
-    gitUri,
-    ...analysisData,
-  };
+  if(!sarif){
+    return {
+      baseURL,
+      sessionToken,
+      includeLint,
+      severity,
+      gitUri,
+      ...analysisData,
+    };
+  }
+  else {
+    return {
+      baseURL,
+      sessionToken,
+      includeLint,
+      severity,
+      gitUri,
+      ...analysisData,
+      sarifResults: analysisData.analysisResults ? await new Sarif(analysisData.analysisResults).sarifConverter() : undefined,
```

Comment:
Why creating a new instance of the class for each conversion? "await" and "new" both have their own computational weight which could be easily avoided by using static functions.